### PR TITLE
feat/add-support-for-user-treasury

### PR DESCRIPTION
**Pull request by marcin-janas**
Requestor/Jira: @marcin-janas / https://jira.airhelp.com/browse/DOT-377
Risk (low/med/high): low
Tested (yes/no): yes
```
$ AWS_PROFILE=staging go run main.go write user/marcin.janas/phone test
Success! Data written to:  user/marcin.janas/phone

$ AWS_PROFILE=staging go run main.go read user/marcin.janas/phone
test
```
Description/Why: we must be able to save the phone for each user in the treasury, and the username with a dot is the default in many places, e.g. chef, AWS, e-mail, so I decided to add exactly the same username pattern to the treasury

### CODE REVIEW

File: utils/utils_test.go [hunk 1]
```diff
@@ -21,6 +23,8 @@ func TestValidateInputKey(t *testing.T) {
 		"45678901jbf",
 		"asasa/123!12/312313",
 		"1231/1231*2/312313",
+		"user/username/phone",
```
Comment (kszarek) [inline utils/utils_test.go:26]: I'm not sure about it but ok.

File: utils/utils.go [hunk 2]
```diff
@@ -12,15 +12,23 @@ import (
...
+	}
 	return validate(cliIn, environmentApplicationKeyRegex)
 }
 
 // ValidateInputKeyPattern checks if cli input is valid, also without key name
 func ValidateInputKeyPattern(cliIn string) error {
+	if strings.HasPrefix(cliIn, "user") {
+		return validate(cliIn, userUsernameRegex)
+	}
 	return validate(cliIn, environmentApplicationRegex)
```
Comment (maciej-wilczynski) [inline utils/utils.go:32]: i don't like it, i think it should be in validate() method but i don't know how

**Comment by maciej-wilczynski** [comment]
The functionality that you created here should be described in readme file :)

File: utils/utils_test.go (hunk 1 above)
Comment (marcin-janas) [reply utils/utils_test.go:26]: I forced this pattern here https://github.com/AirHelp/treasury/pull/106/files#diff-cd575165f84eed10d10eb1460b0927b4R15-R16
so this is not a valid string

File: utils/utils.go (hunk 2 above)
Comment (marcin-janas) [reply utils/utils.go:32]: according to the conversation, we don't change it

**Comment by marcin-janas** [comment]
I added a section in README... pls look and re-approve :)
https://github.com/AirHelp/treasury/tree/feat/add-support-for-user-treasury#treasury-as-a-user-vault

**Pull request merged**